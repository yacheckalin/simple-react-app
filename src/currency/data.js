export const currencyData = [
  {
    favorite: false,
    asset: "EUR/USD",
    quote: "1.12",
    date: new Date("2019-04-08"),
  },
  {
    favorite: false,
    asset: "USD/JPY",
    quote: "107.95",
    date: new Date("2019-04-11"),
  },
  {
    favorite: false,
    asset: "GBP/USD",
    quote: "1.238",
    date: new Date("2019-04-18"),
  },
  {
    favorite: false,
    asset: "AUD/USD",
    quote: "0.7048",
    date: new Date("2020-04-28"),
  },
  {
    favorite: false,
    asset: "USD/CHF",
    quote: "0.9888",
    date: new Date("2020-04-08"),
  },
  {
    favorite: false,
    asset: "USD/CAD",
    quote: "1.31",
    date: new Date("2019-02-08"),
  },
];
