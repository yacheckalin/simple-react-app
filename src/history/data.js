export const data = [
  {
    asset: "USD/JPD",
    startDate: new Date("2019-04-08 04:30:00"),
    startQuote: "1.12222",
    finishDate: new Date("2019-04-08 06:30:00"),
    finishQuote: "1.2312",
    profit: "+1",
  },
  {
    asset: "CAD/JPD",
    startDate: new Date("2019-02-08 06:30:00"),
    startQuote: "1.12222",
    finishDate: new Date("2019-04-08 06:30:00"),
    finishQuote: "1.2312",
    profit: "+10.08",
  },
  {
    asset: "CAD/USD",
    startDate: new Date("2019-03-08 06:30:00"),
    startQuote: "1.12222",
    finishDate: new Date("2019-03-08 05:30:00"),
    finishQuote: "1.2312",
    profit: "-10.08",
  },
  {
    asset: "USD/CAD",
    startDate: new Date("2019-03-08 06:30:00"),
    startQuote: "1.12222",
    finishDate: new Date("2020-05-08 05:30:00"),
    finishQuote: "1.2312",
    profit: "4",
  },
  {
    asset: "USD/CAD",
    startDate: new Date("2019-03-08 06:30:00"),
    startQuote: "1.12222",
    finishDate: new Date("2020-05-08 05:30:00"),
    finishQuote: "1.2312",
    profit: "4",
  },
  {
    asset: "USD/CAD",
    startDate: new Date("2019-03-08 06:30:00"),
    startQuote: "1.12222",
    finishDate: new Date("2020-05-08 05:30:00"),
    finishQuote: "1.2312",
    profit: "4",
  },
  {
    asset: "USD/CAD",
    startDate: new Date("2019-03-08 06:30:00"),
    startQuote: "1.12222",
    finishDate: new Date("2020-05-08 05:30:00"),
    finishQuote: "1.2312",
    profit: "4",
  },
  {
    asset: "GBP/USD",
    startDate: new Date("2019-03-08 06:30:00"),
    startQuote: "1.12222",
    finishDate: new Date("2019-03-08 02:30:00"),
    finishQuote: "1.2312",
    profit: "-0.08",
  },
  {
    asset: "USD/JPD",
    startDate: new Date("2019-04-08 04:30:00"),
    startQuote: "1.12222",
    finishDate: new Date("2019-04-08 06:30:00"),
    finishQuote: "1.2312",
    profit: "+0.08",
  },
  {
    asset: "CAD/JPD",
    startDate: new Date("2019-02-08 06:30:00"),
    startQuote: "1.12222",
    finishDate: new Date("2019-04-08 06:30:00"),
    finishQuote: "1.2312",
    profit: "-1.08",
  },
  {
    asset: "CAD/USD",
    startDate: new Date("2019-03-08 06:30:00"),
    startQuote: "1.12222",
    finishDate: new Date("2019-03-08 05:30:00"),
    finishQuote: "1.2312",
    profit: "-10.08",
  },
  {
    asset: "USD/CAD",
    startDate: new Date("2019-03-08 06:30:00"),
    startQuote: "1.12222",
    finishDate: new Date("2019-05-08 05:30:00"),
    finishQuote: "1.2312",
    profit: "1.08",
  },
  {
    asset: "GBP/USD",
    startDate: new Date("2019-03-08 06:30:00"),
    startQuote: "1.12222",
    finishDate: new Date("2019-03-08 02:30:00"),
    finishQuote: "1.2312",
    profit: "-0.08",
  },
  {
    asset: "USD/JPD",
    startDate: new Date("2019-04-08 04:30:00"),
    startQuote: "1.12222",
    finishDate: new Date("2019-04-08 06:30:00"),
    finishQuote: "1.2312",
    profit: "+0.08",
  },
  {
    asset: "CAD/JPD",
    startDate: new Date("2019-02-08 06:30:00"),
    startQuote: "1.12222",
    finishDate: new Date("2019-04-08 06:30:00"),
    finishQuote: "1.2312",
    profit: "+10.08",
  },
  {
    asset: "CAD/USD",
    startDate: new Date("2019-03-08 06:30:00"),
    startQuote: "1.12222",
    finishDate: new Date("2019-03-08 05:30:00"),
    finishQuote: "1.2312",
    profit: "-10.08",
  },
  {
    asset: "USD/CAD",
    startDate: new Date("2019-03-08 06:30:00"),
    startQuote: "1.12222",
    finishDate: new Date("2019-05-08 05:30:00"),
    finishQuote: "1.2312",
    profit: "1.08",
  },
  {
    asset: "GBP/USD",
    startDate: new Date("2019-03-08 06:30:00"),
    startQuote: "1.12222",
    finishDate: new Date("2019-03-08 02:30:00"),
    finishQuote: "1.2312",
    profit: "-0.08",
  },
  {
    asset: "GBP/USD",
    startDate: new Date("2018-03-08 06:30:00"),
    startQuote: "1.12222",
    finishDate: new Date("2018-02-08 02:30:00"),
    finishQuote: "1.2312",
    profit: "102.08",
  },
  {
    asset: "GBP/USD",
    startDate: new Date("2019-03-08 06:30:00"),
    startQuote: "1.12222",
    finishDate: new Date("2018-03-08 02:30:00"),
    finishQuote: "1.2312",
    profit: "100.08",
  },
];
