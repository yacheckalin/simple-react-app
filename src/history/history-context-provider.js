import React, { useContext, useState, useEffect } from "react";

import { data } from "./data";
import { INITIAL_PAGINATION_PAGE, DEFAULT_PAGINATION_LIMIT } from "./constants";
import { customSort } from "./helpers";

export const HistoryContext = React.createContext();
export const useHistoryContext = () => useContext(HistoryContext);

const HistoryContextProvider = ({ children }) => {
  const [paginationList, setPaginationList] = useState([]);
  const [currentPage, setCurrentPage] = useState(INITIAL_PAGINATION_PAGE);
  const [totalPages, setTotalPages] = useState(0);

  useEffect(() => {
    //TODO: change this part for axios request to API
    const successResponse = { result: "ok", deals: data };
    const fetchData = () => Promise.resolve(successResponse);
    let paginated = [];

    const paginate = (data) => {
      return data.splice(
        currentPage * DEFAULT_PAGINATION_LIMIT,
        DEFAULT_PAGINATION_LIMIT
      );
    };

    fetchData().then((response) => {
      if (response.result === "ok") {
        const sorted = customSort(response.deals, currentPage);
        paginated = paginate(sorted);
        setTotalPages(
          Math.floor(response.deals.length / DEFAULT_PAGINATION_LIMIT)
        );

        setPaginationList(paginated);
      }
    });
    return () => setPaginationList(paginated);
  }, [currentPage]);

  const backward = () => {
    setCurrentPage(currentPage - 1);
  };

  const forward = () => {
    setCurrentPage(currentPage + 1);
  };

  return (
    <HistoryContext.Provider
      value={{
        data: paginationList,
        backward,
        forward,
        currentPage,
        totalPages,
      }}
    >
      {children}
    </HistoryContext.Provider>
  );
};

export default HistoryContextProvider;
