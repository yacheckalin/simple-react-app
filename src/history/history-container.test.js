import React from "react";
import { shallow, mount, render } from "enzyme";
import toJSON from "enzyme-to-json";

import HistoryContainer from "./history-container";
import * as HistoryContext from "./history-context-provider";
import { data } from "./data";

describe("<HistoryContainer />", () => {
  test("renders and displays properly", () => {
    const contextValues = {
      data,
    };
    jest
      .spyOn(HistoryContext, "useHistoryContext")
      .mockImplementation(() => contextValues);
    const wrapper = shallow(<HistoryContainer />);
    expect(toJSON(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
