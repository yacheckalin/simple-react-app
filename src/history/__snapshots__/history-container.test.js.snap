// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HistoryContainer /> renders and displays properly 1`] = `
<styled.table>
  <thead>
    <tr>
      <th>
         
      </th>
      <th>
        Актив
      </th>
      <th>
        Начало
      </th>
      <th>
        Котировка
      </th>
      <th>
        Конец
      </th>
      <th>
        Котировка
      </th>
      <th>
        Прибыль
      </th>
      <th>
         
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="0"
    >
      <td>
         
      </td>
      <td>
        USD/JPD
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.12222
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.2312
      </td>
      <td>
        +1
      </td>
      <td>
         
      </td>
    </tr>
    <tr
      key="1"
    >
      <td>
         
      </td>
      <td>
        CAD/JPD
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.12222
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.2312
      </td>
      <td>
        +10.08
      </td>
      <td>
         
      </td>
    </tr>
    <tr
      key="2"
    >
      <td>
         
      </td>
      <td>
        CAD/USD
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.12222
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.2312
      </td>
      <td>
        -10.08
      </td>
      <td>
         
      </td>
    </tr>
    <tr
      key="3"
    >
      <td>
         
      </td>
      <td>
        USD/CAD
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.12222
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.2312
      </td>
      <td>
        4
      </td>
      <td>
         
      </td>
    </tr>
    <tr
      key="4"
    >
      <td>
         
      </td>
      <td>
        USD/CAD
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.12222
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.2312
      </td>
      <td>
        4
      </td>
      <td>
         
      </td>
    </tr>
    <tr
      key="5"
    >
      <td>
         
      </td>
      <td>
        USD/CAD
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.12222
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.2312
      </td>
      <td>
        4
      </td>
      <td>
         
      </td>
    </tr>
    <tr
      key="6"
    >
      <td>
         
      </td>
      <td>
        USD/CAD
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.12222
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.2312
      </td>
      <td>
        4
      </td>
      <td>
         
      </td>
    </tr>
    <tr
      key="7"
    >
      <td>
         
      </td>
      <td>
        GBP/USD
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.12222
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.2312
      </td>
      <td>
        -0.08
      </td>
      <td>
         
      </td>
    </tr>
    <tr
      key="8"
    >
      <td>
         
      </td>
      <td>
        USD/JPD
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.12222
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.2312
      </td>
      <td>
        +0.08
      </td>
      <td>
         
      </td>
    </tr>
    <tr
      key="9"
    >
      <td>
         
      </td>
      <td>
        CAD/JPD
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.12222
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.2312
      </td>
      <td>
        -1.08
      </td>
      <td>
         
      </td>
    </tr>
    <tr
      key="10"
    >
      <td>
         
      </td>
      <td>
        CAD/USD
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.12222
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.2312
      </td>
      <td>
        -10.08
      </td>
      <td>
         
      </td>
    </tr>
    <tr
      key="11"
    >
      <td>
         
      </td>
      <td>
        USD/CAD
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.12222
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.2312
      </td>
      <td>
        1.08
      </td>
      <td>
         
      </td>
    </tr>
    <tr
      key="12"
    >
      <td>
         
      </td>
      <td>
        GBP/USD
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.12222
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.2312
      </td>
      <td>
        -0.08
      </td>
      <td>
         
      </td>
    </tr>
    <tr
      key="13"
    >
      <td>
         
      </td>
      <td>
        USD/JPD
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.12222
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.2312
      </td>
      <td>
        +0.08
      </td>
      <td>
         
      </td>
    </tr>
    <tr
      key="14"
    >
      <td>
         
      </td>
      <td>
        CAD/JPD
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.12222
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.2312
      </td>
      <td>
        +10.08
      </td>
      <td>
         
      </td>
    </tr>
    <tr
      key="15"
    >
      <td>
         
      </td>
      <td>
        CAD/USD
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.12222
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.2312
      </td>
      <td>
        -10.08
      </td>
      <td>
         
      </td>
    </tr>
    <tr
      key="16"
    >
      <td>
         
      </td>
      <td>
        USD/CAD
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.12222
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.2312
      </td>
      <td>
        1.08
      </td>
      <td>
         
      </td>
    </tr>
    <tr
      key="17"
    >
      <td>
         
      </td>
      <td>
        GBP/USD
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.12222
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.2312
      </td>
      <td>
        -0.08
      </td>
      <td>
         
      </td>
    </tr>
    <tr
      key="18"
    >
      <td>
         
      </td>
      <td>
        GBP/USD
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.12222
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.2312
      </td>
      <td>
        102.08
      </td>
      <td>
         
      </td>
    </tr>
    <tr
      key="19"
    >
      <td>
         
      </td>
      <td>
        GBP/USD
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.12222
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD HH:mm:ss"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
        1.2312
      </td>
      <td>
        100.08
      </td>
      <td>
         
      </td>
    </tr>
  </tbody>
  <tfoot>
    <tr>
      <td
        colSpan={8}
      >
        <Pagination />
      </td>
    </tr>
  </tfoot>
</styled.table>
`;
